apps/guild-import-job/src/nats.ts

```diff
@@ -0,0 +1,148 @@
+import {
+  AckPolicy,
+  connect,
+  JsMsg,
+  nanos,
+  NatsConnection,
+  StringCodec,
+} from 'nats';
+
+// TODO: How are streams created
+// TODO: How is pulling filtered
+
+/**
+ * A helper function to setup a connection to NATS and create a stream to upload messages,
+ * trying to connect to each server in the `servers` array.
+ * Once connected, it creates a stream with the specified `streamName`.
+ *
+ * @param servers - an array of servers to connect.
+ * @param streamName - the name (subject) of the stream to create.
+ * @param work - a callback function to perform  further actions with the connection.
+ */
+export const setupNats = async (
+  servers: { servers?: string; port?: number }[],
+  streamName: string,
+  // NOTE: is async needed here?
+  work: (conn: NatsConnection) => Promise<void>,
+) => {
+  for (const v of servers) {
+    try {
+      const nc = await connect(v);
+      console.log(`connected to ${nc.getServer()}`);
+
+      // create a stream to upload messages
+      const jsm = await nc.jetstreamManager();
+      const subj = `${streamName}.*`;
+      const streamCfg = await jsm.streams.add({
+        name: streamName,
+        subjects: [subj],
+      });
+      console.log(`created stream ${streamCfg}`);
+
+      // this promise indicates the client closed
+      const done = nc.closed();
+      // do something with the connection
+      await work(nc);
+
+      // close the connection
+      await nc.close();
+      // check if the close was OK
+      const err = await done;
+      if (err) {
+        console.log(`error closing:`, err);
+      }
+    } catch (err) {
+      console.log(err);
+      console.log(`error connecting to ${JSON.stringify(v)}`);
+    }
+  }
+};
+
+/**
+ * A helper function to write messages to a stream, creating a `JetStreamClient`
+ * and publishing given `messages` to it.
+ *
+ * @param nc - the connection to use.
+ * @param streamName - the name of the stream to write to.
+ * @param messages
+ */
+export const writeMessages = async (
+  nc: NatsConnection,
+  streamName: string,
+  messages: string[],
+) => {
+  const js = nc.jetstream();
+  const sc = StringCodec();
+  console.log(`:: GUILDDDDDD: ${messages}`);
+
+  for (const m of messages) {
+    console.log(`:: GUILDDDDDD: ${m}`);
+
+    // TODO: What if the stream name/subject doesn't exist?
```

Comment:
Did you test this? What is the answer?

Reply:
yeah, it throws this error: 
`Error: no stream matches subject`.